src/LibraryManager/CacheService/CacheService.cs

```diff
@@ -124,28 +126,20 @@ private async Task<string> GetResourceAsync(string url, string localFile, int ex
         }
 
         /// <summary>
-        /// Refreshes the cache for the given set of files if expired 
+        /// Refreshes the cache for the given set of files if expired
         /// </summary>
-        /// <param name="librariesCacheMetadata"></param>
-        /// <param name="cancellationToken"></param>
-        /// <returns></returns>
-        public async Task RefreshCacheAsync(IEnumerable<CacheFileMetadata> librariesCacheMetadata, CancellationToken cancellationToken)
+        public async Task RefreshCacheAsync(IEnumerable<CacheFileMetadata> librariesCacheMetadata, ILogger logger, CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
-
-            List<Task> refreshTasks = new List<Task>();
+            await ParallelUtility.ForEachAsync(DownloadFileIfNecessaryAsync, MaxConcurrentDownloads, librariesCacheMetadata, cancellationToken);
 
-            foreach (CacheFileMetadata metadata in librariesCacheMetadata)
+            async Task DownloadFileIfNecessaryAsync(CacheFileMetadata metadata)
             {
-                if (!File.Exists(metadata.DestinationPath) || File.GetLastWriteTime(metadata.DestinationPath) < DateTime.Now.AddDays(-_libraryExpiresAfterDays))
+                if (!File.Exists(metadata.DestinationPath) || File.GetLastWriteTime(metadata.DestinationPath) < DateTime.Now.AddDays(-LibraryExpiresAfterDays))
```

Comment:
Might be nice if instead of  DateTime.Now.AddDays(-LibraryExpiresAfterDays), LibraryExpiresAfterDays was rather a TimeSpan and we could do DateTime.Now.Subtract instead.

Reply:
Next cleanup item on my list is to remove this expiration anyway.  Package contents should be static for a given version, so it seems dumb to make you redownload them every 30 days just to get the same file contents back.

(Unless you're modifying files in the cache, in which case it's on your own head to restore or remove them.)

Reply:
Bah, I might as well just do this now, it'll also remove the Utc comment below. ;P